### DISCUSSION TITLE
Welcome to Dolphin_Emulator_RVZ_Utility_Scripts Discussions!

### DISCUSSION THREAD

**Discussion by ElektroStudios** [ElektroStudios/Dolphin_Emulator_RVZ_ISO_GameCube_Wii_Conversion_Scripts#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by crvylooo** [comment]
Hi there, does this script work on Wii isos as well? Or is it for Gamecube only

**Reply by ElektroStudios** [reply]
Hi and thanks for participating!. Yes, **DolphinTool** supports GameCube and Wii ISO files, so the script files will work for Wii ISO files too. 

I've published a new release, please check it.

Note: try changing `$compressionLevel` value to `8` (instead of `9`) in the script file for Wii ISO files if the script (better said, if DolphinTool.exe) throws any conversion error.

**Reply by crvylooo** [reply]
Thank you for your quick response, I really appreciate it. I haven't gotten a chance to use the tool yet as my gamecube isos are still downloading (my library was all ciso files for gc and wbfs for wii but I plan to save more space by going to rvz)
I'm fairly stupid when it comes to code execution, especially when it's not handled by a nice little gui, so I just wanted to make sure; I put all my iso files in the same folder as the Convert_ISO_to_RVZ.ps1 script file, then just execute the script file right?

**Reply by ElektroStudios** [reply]
Exactly, it's that simple: put everything together in the same directory, then run the conversion script (the one that has .ps1 file extension rather than the .bat file).

The script is configured to generate RVZ files using the maximum allowed compression parameters.

Maybe when I have time I will develop a graphical user interface.

Sincerely,
Elektro.

**Reply by crvylooo** [reply]
With such simple instructions, one wouldn't even be necessary. Thanks again. I plan on using these files to play on my Steam Deck.

I just realized my wii isos are "nkit.iso" files. Will this affect the conversion?

**Reply by ElektroStudios** [reply]
> With such simple instructions, one wouldn't even be necessary. Thanks again. I plan on using these files to play on my Steam Deck.
> 
> I just realized my wii isos are "nkit.iso" files. Will this affect the conversion?

I'm not sure about the **NKit** format, I've never used it, but from what I think that I understand: `NKIT.ISO` files are scrubbed ISO files with the **NKit** tool. They are trimmed ISO files from what I understand. On the other hand, `NKit.GCZ` files are compressed files with the deprecated **NKit** format, and at least I can say that any `NKIT.GCZ` file for sure it must be reverted to ISO file format in order to compress them with **RVZ** format.

But I don't know if a `NKIT.ISO` file will work with **DolphinTool**; maybe the tool will expect some partition that was scrubbed from the original, full-sized ISO. You're going to have to try it.

You can find **Nkit** available to download here: https://wiki.gbatemp.net/wiki/NKit

But there is a chance that you may find a problem reverting **Nkit** GCZ/ISO files to full-sized ISO files; see the "Can be reverted?" commentary from this quote:

> NKit formats - GC/Wii
> 
> This format has been deprecated in favor of [RVZ](https://emulation.gametechwiki.com/index.php/Save_disk_space_for_ISOs#RVZ_.28Modern_Dolphin_format.29_-_GC.2FWii). ([Developer's Wiki](https://github.com/Nanook/NKit/wiki))
> 
>     Archive-quality dump? Yes (Not for Wii games, if the Wii update partition is removed and the recovery file not preserved)
>     Gain: Immediate
>     Tools Used: NKit
>     Can be reverted? Yes (unless the Wii update partition was removed and the recovery file not preserved)
>     Playable on Hardware? GC: Yes (ISO variant only) / Wii: No
>     Playable on Emulators? Yes (Dolphin)
> 
> NKit.GCZ is less space efficient than NKit.ISO compressed with a powerful general purpose archiver (7zip), however it is directly playable in Dolphin.

Source: https://emulation.gametechwiki.com/index.php/Save_disk_space_for_ISOs#NKit_formats_-_GC.2FWii

**Reply by crvylooo** [reply]
Yeah it looks like nkit is failing the verification for these nkits and will not output them to iso... I am having trouble finding an archive of just iso files unfortunately for wii games so I'm unsure how to proceed, without going and downloading my library 1 by 1 from archives.

**Reply by crvylooo** [reply]
I found a wbfs to iso converter! It can only convert 1 by 1 lol, but at least I've got a easy way to get isos to use this for.
I just did all of my gamecube games. I noticed it says a message "this is not optimal for usage" (sorry I did not capture the exact message!) this wont give me errors and long load times when I play them on my steam deck with dolphin, will it?

**Reply by ElektroStudios** [reply]
> "this is not optimal for usage" (sorry I did not capture the exact message!) this wont give me errors and long load times when I play them on my steam deck with dolphin, will it?

The warning message is expected when using a value greater than 2MB block size; it can be ignored. Using a greater value should not give you errors nor increase load times, I've played hundreds of (GameCube) ISO files converted to RVZ this way, and didn't found any issue.

This is something I wrote time ago on other site:

> From the tests I did playing many games in RVZ format using different block sizes, I'm pretty sure that using larger block sizes above 2 MB IT DOES NOT AFFECT EMULATION PERFORMANCE OR LOAD TIMES.

**Reply by crvylooo** [reply]
Hi again. I have tested out a few games on my steam deck, and I have to admit this does seem to affect performance greatly. A ton of stuttering, especially in cutscenes, and very long load times. What would be a better block size?

**Reply by ElektroStudios** [reply]
Wow, I'm sorry to hear that! I've only tried it on PC, where I can say that I've cautiously measured the loading screen times, and the FPS, and there's no noticeable difference, and no stuttering; the games I've tried are as responsive as if I play them without compression. I have a PC that is eight years old (AMD Ryzen 7 1700X, nVidia GTX 1060).

Thank you for describing the problem you have suffered, these kind of personal experiences can help us better understand why the **Dolphin** developers decided to show a warning when using dictionary sizes larger than 2MB. Although, as I said, on my PC there is no difference in the emulator/game response regardless of the dictionary size used for RVZ compression.

Answering to your question, open the **ISO to RVZ** script file and change the specified dictionary size of 32mb:
`$dictionarySize    = 32mb`

To the official maximum suggested value by **Dolphin** developers which is 2mb:
`$dictionarySize    = 2mb`

That is the maximum dictionary size suggested, but If you want to know, the default value in **DolphinTool** is 128kb. The script file contains a commentary block with the default values in case of you want to try them:

```
Default Dolphin values:
-----------------------
$compressionFormat = "zstd"
$compressionLevel  = 5
$dictionarySize    = 128kb
```

But at the end all this depends on the hardware capabilities of the Steam Deck. If the Steam Deck still giving issues in the emulation, try lowering the dictionary size, and/or try using **zstd** compression instead for theoretically faster reading times: 

`$compressionFormat = "zstd"`

**Reply by crvylooo** [reply]
Great, I'll try this out now! I can use this script to reconvert all my files back to ISO with no need to redownload any roms, right?

**Reply by ElektroStudios** [reply]
Yes, of course, use the [Extract_RVZ_to_ISO.ps1](https://github.com/ElektroStudios/Dolphin_Emulator_RVZ_ISO_GameCube_Wii_Conversion_Scripts/blob/main/Source/PowerShell/Extract_RVZ_to_ISO.ps1) script file to revert compressed RVZ files back to ISO format.

**Reply by crvylooo** [reply]
I've changed the script's settings to dolphin's default (128kb & zstd compression format) but the cmd still displays the "Block size is still not ideal for performance" error, why is that?
Do I need to just edit this section of the code
`ECho:[Compression=zstd, Compression Level=9, Block Size=128kb]`
Or this one as well?
`	"DolphinTool.exe" convert --format=rvz --input="%%~f#" --output="%%~n#.rvz" ^
	                          --block_size=33554432 --compression=zstd --compression_level=9`
